tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -0,0 +1,114 @@
+        'Average': Math.mean,
+        'Total': Math.sum
+    }
+
+    def __init__(self, data, manager=None):
+        super(MetricFilter, self).__init__(data, manager)
+        # Metric name as defined by Azure SDK
+        self.metric = self.data.get('metric')
+        # gt (>), ge (>=), eq (==), le (<=), lt (<)
+        self.op = self.data.get('op')
+        # Value to compare metric value with self.op
+        self.threshold = self.data.get('threshold')
+        # Number of hours from current UTC time
+        self.timeframe = float(self.data.get('timeframe', self.DEFAULT_TIMEFRAME))
+        # Interval as defined by Azure SDK
+        self.interval = self.data.get('interval', self.DEFAULT_INTERVAL)
+        # Aggregation as defined by Azure SDK
+        self.aggregation = self.data.get('aggregation', self.DEFAULT_AGGREGATION)
+        # Aggregation function to be used locally
+        self.func = self.aggregation_funcs[self.aggregation]
+
+    def validate(self):
+        # Give appropriate error messages for missing information
+        if self.metric is None:
+            raise PolicyValidationError("Need to define a metric")
+        if self.func is None:
+            raise PolicyValidationError("Need to define a func (avg, min, max)")
+        if self.op is None:
+            raise PolicyValidationError("Need to define an opeartor (gt, ge, eq, le, lt)")
+        if self.threshold is None:
+            raise PolicyValidationError("Need to define a threshold")
+        try:
+            self.threshold = float(self.threshold)
+        except ValueError:
+            raise PolicyValidationError("Threshold needs to be a valid number")
+
+    def process(self, resources, event=None):
+        # Import utcnow function as it may have been overridden for testing purposes
+        from c7n_azure.actions import utcnow
+
+        # Get timespan
+        end_time = utcnow()
+        start_time = end_time - timedelta(hours=self.timeframe)
+        self.timespan = "{}/{}".format(start_time, end_time)
+
+        # Create Azure Monitor client
+        self.client = self.manager.get_client('azure.mgmt.monitor.MonitorManagementClient')
+
+        # Process each resource in a separate thread, returning all that pass filter
+        with self.executor_factory(max_workers=3) as w:
+            processed = list(w.map(self.process_resource, resources))
+            return [item for item in processed if item is not None]
+
+    def get_metric_data(self, resource):
+        metrics_data = self.client.metrics.list(
+            resource['id'],
+            timespan=self.timespan,
+            interval=self.interval,
+            metric=self.metric,
+            aggregation=self.aggregation
+        )
+        return [item.total for item in metrics_data.value[0].timeseries[0].data]
+
+    def passes_op_filter(self, resource):
+        m_data = self.get_metric_data(resource)
+        aggregate_value = self.func(m_data)
+
+        if self.op == 'ge':
```

Comment:
this logic tends to be simpler as symbolic to function map lookup. ie `ops = {'le': operator.le}`, and then `return  ops[self.op](aggregate_value, threshold)` in fact we already have that mapping defined in the value filter.